### PR TITLE
Add adapter for Either<ErrorBody, ResponseBody>

### PR DESCRIPTION
**Pull request by LordRaydenMK**
Supports declaring your Retrofit interface like:

```
interface SuspedApiClientTest {

  @GET("/")
  suspend fun getEither(): Either<ErrorMock, ResponseMock>

  @GET("/")
  suspend fun getResponseE(): ResponseE<ErrorMock, ResponseMock>
}
```
note: `ResponseE` is the equivalent of `retrofit.Response` but using Either for the body (left in case of HTTP error, right in case of success).

Problems:
There is a null check for the body, meaning it errors on 204 (No Content) because the body is null in that case.

### CODE REVIEW

File: arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/CallKindAdapterFactory.kt [hunk 1]
```diff
@@ -1,36 +1,4 @@
...
-    }
-  }
-}
-
-private fun parseTypeName(type: Type) =
-  type.toString()
-    .split(".")
-    .last()
+@Deprecated("Please use ArrowCallAdapterFactory instead", ReplaceWith("ArrowCallAdapterFactory"))
+typealias CallKindAdapterFactory = ArrowCallAdapterFactory
```
Comment (LordRaydenMK) [inline arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/CallKindAdapterFactory.kt:4]: IMO doesn't make sense to have multiple factories, so I named the one true factory `ArrowCallAdapterFactory`. Old name is now a typealis and deprecated.

**Comment by LordRaydenMK** [comment]
hey @leandroBorgesFerreira 

I extracted a function in `ArrowCallAdapterFactory` for the duplicated code.

For the `ArrowEitherCallAdapter` and `ArrowEitherECallAdapter` the code looks similar, but the types are all different e.g. `Call<Either<E, R>>` VS `Call<ResponseE<E, R>>` the `Callback` type is also different, so I would have to parametrize a lot of things. That would make the code harder to read IMO.

File: arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/CallKindAdapterFactory.kt (hunk 1 above)
Comment (LordRaydenMK) [reply arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/CallKindAdapterFactory.kt]: Reverted this change.

File: arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/either/EitherCallAdapterFactory.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  override fun get(returnType: Type, annotations: Array<Annotation>, retrofit: Retrofit): CallAdapter<*, *>? {
+    val rawType = getRawType(returnType)
+
+    if (returnType !is ParameterizedType) {
+      val name = parseTypeName(returnType)
+      throw IllegalArgumentException("Return type must be parameterized as " +
+        "$name<Foo> or $name<out Foo>")
+    }
+
+    val effectType = getParameterUpperBound(0, returnType)
```
Comment (aballano) [inline arrow-integrations-retrofit-adapter/src/main/kotlin/arrow/integrations/retrofit/adapter/either/EitherCallAdapterFactory.kt:45]: this variable doesn't seem to be used

```suggestion
```

**Pull request merged**